src/app/user/users.component.ts

```diff
@@ -1,28 +1,47 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
 import { User } from '../../backend/model/index';
 import { UsersService } from '../../backend/services/index';
+import { ToastrService } from 'ngx-toastr';
+import { DatatableComponent } from '@swimlane/ngx-datatable';
 
 @Component({
     selector: 'app-users',
     templateUrl: './users.component.html',
     styleUrls: ['./users.component.scss']
 })
 export class UsersComponent implements OnInit {
+    @ViewChild('actionTmpl') actionTmpl: TemplateRef<any>;
+    @ViewChild('enabledTmpl') enabledTmpl: TemplateRef<any>;
+    @ViewChild(DatatableComponent) table: DatatableComponent;
+
     users: User[] = [];
+    rows: User[] = [];
+    columns = [];
 
     constructor(
-        private usersService: UsersService
+        private usersService: UsersService,
+        private toastr: ToastrService
     ) {
     }
 
     ngOnInit() {
-        this.usersService.getAll().subscribe(users => this.users = users);
+        this.usersService.getAll().subscribe(users => this.rows = this.users = users);
+
+        this.columns = [
+            {prop: 'firstName', name: 'Prénom'},
+            {prop: 'lastName', name: 'Nom'},
+            {prop: 'profile.label', name: 'Profil'},
+            {prop: 'enabled', name: 'Statut', cellTemplate: this.enabledTmpl},
+            {name: 'Actions', cellTemplate: this.actionTmpl},
+        ];
     }
 
     delete(user: User) {
         if (confirm('Etes-vous sûr de vouloir supprimer la ligne sélectionnée ?')) {
             this.usersService.remove(user).subscribe(() => {
+                this.toastr.success(`${user.fullName} a bien été supprimé 🗑`)
                 this.users.splice(this.users.indexOf(user), 1);
```

Comment:
Ce splice la ne sert plus je pense vu que tu le fais sur le this.rows

Reply:
si ça l'est parce que le this.rows n'est que les users filtrés donc faut maintenir les 2 à jours